run_unittests.py

```diff
@@ -3232,15 +3235,52 @@ def dist_impl(self, vcs_init, vcs_add_all=None, include_subprojects=True):
             if vcs_add_all:
                 vcs_add_all(self.create_dummy_subproject(project_dir, 'samerepo'))
             self.init(project_dir)
-            self.build('dist')
-            self.assertPathExists(xz_distfile)
-            self.assertPathExists(xz_checksumfile)
-            self.assertPathDoesNotExist(zip_distfile)
-            self.assertPathDoesNotExist(zip_checksumfile)
-            self._run(self.meson_command + ['dist', '--formats', 'zip'],
-                      workdir=self.builddir)
-            self.assertPathExists(zip_distfile)
-            self.assertPathExists(zip_checksumfile)
+            self.build()
+
+            for i in range(len(archive_choices) + 1):
+                permutations = tuple(itertools.permutations(archive_choices, i))
+                for i, subset in enumerate(permutations):
```

Comment:
Is it really worth testing all possible permutations? I would just test the case where you have all archives with `','.join(archive_choices)`.

Reply:
Good point. I think will do as you have said.

Reply:
At least one test that isn't all of the options is probably useful. I've been bitten enough times by "but I tested x + y + z and it worked, why doesn't just x wor‽"